### ISSUE TITLE
Idea for an FSI example: Write an adapter for WaterLily.jl

### ISSUE THREAD

**Issue by MakisH** [precice/PreCICE.jl#11]
[WaterLily.jl](https://github.com/weymouth/WaterLily.jl) seems to be a very easy to use flow solver in Julia: https://julialang.org/blog/2021/08/sharks/

Writing an adapter for this would give us a first tutorial case where we could demonstrate the Julia bindings.

(I have no idea if there are more mature flow solvers in Julia, I just discovered this and it looks easy and interesting.

**Comment by uekerman** [comment]
After a quick first look: It seems the code uses a fixed Cartesian mesh. If moving boundaries are done similarly as in the "CFD-Lab code", FSI with preCICE will be complicated to do.
Easier could be to extend the code with heat transport and go for CHT. 
Or directly for fluid-fluid coupling.

**Comment by dynamic-queries** [comment]
https://github.com/trixi-framework/Trixi.jl has some nice solvers and examples for PDEs with conservation laws. 
This could be looked into. 

WaterLily.jl is not a conventional solver per se. It is a data generation package for Deep Learning models. And the way they approximate the vector field is really dubious. See https://www.youtube.com/watch?v=YsPkfZqbNSQ , for an intro into the WaterLilly.jl package.

**Comment by uekerman** [comment]
@marinlauber Definitely. We currently have no Julia-based solver in our tutorials. Anything there to continuously test the Julia bindings could be very helpful.
I have no idea, however, how complicated this would be.
Who is Waterlily's bigger brother? :grin:

**Comment by marinlauber** [comment]
@uekerman Sounds good! I already have the preCICE coupling working for WaterLily, but for a simple IGA solver. Together with @hverhelst we are actually looking at binding WaterLily.jl and G+Smo to preCICE and running some example cases.
WaterLily's big brother is our in-house fortran solver, it's essentially the same solver, but it uses openmpi and has some additional capabilities. Unfortunatly, it's not open-source but, ideally, WaterLily with multi-GPU will replace it at some point.
